utils/terrascript_client.py

```diff
@@ -2271,25 +2280,44 @@ def init_values(self, resource, namespace_info):
         self.override_values(values, overrides)
         values['identifier'] = identifier
         values['tags'] = self.get_resource_tags(namespace_info)
-        values['variables'] = variables
-        values['policies'] = policies
-        values['user_policy'] = user_policy
-        values['region'] = region
-        values['availability_zone'] = az
-        values['queues'] = queues
-        values['specs'] = specs
-        values['parameter_group'] = parameter_group
-        values['sqs_identifier'] = sqs_identifier
-        values['s3_events'] = s3_events
-        values['bucket_policy'] = bucket_policy
-        values['storage_class'] = sc
-        values['enhanced_monitoring'] = enhanced_monitoring
-        values['replica_source'] = replica_source
-        values['es_identifier'] = es_identifier
-        values['filter_pattern'] = filter_pattern
-        values['secret'] = secret
-        values['output_resource_db_name'] = output_resource_db_name
-        values['reset_password'] = reset_password
+        if variables is not None:
```

Comment:
why not
```
if variables:
```
In fact I don't think you need to initialize to `None` in 2256, e.g
```
>>> h = {'a':1}
>>> c = h.get('c')
>>> type(c)
<class 'NoneType'>
>>> if not c: print("Hello Maor")
...
Hello Maor
```

Reply:
in some cases we want to pass an empty string, which fails the `if var` check

Reply:
if we don't set a certain key is `values` we can't get it with `[]`. that is the relation. there is nothing smart here :)

Reply:
ok, understood. I would add a note stating why you want to check explicitly for `None` :)